R/provider-azure.R

```diff
@@ -123,7 +127,23 @@ method(chat_request, ProviderAzure) <- function(provider,
   req <- req_retry(req, max_tries = 2)
   req <- req_error(req, body = function(resp) {
     error <- resp_body_json(resp)$error
-    paste0(error$code, ": ", error$message)
+    msg <- paste0(error$code, ": ", error$message)
+    # Try to be helpful in the (common) case that the user or service
+    # principal is missing the necessary role.
+    # See: https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/role-based-access-control
+    if (error$message == "Principal does not have access to API/Operation.") {
```

Comment:
Maybe use `identical()` here? Otherwise if `message` is every anything other than a string, you'll get a weird error.